src/compare/vanilla.h

```diff
@@ -50,13 +51,60 @@ void VanillaCalcBondsIdx(const T *coords, std::size_t *idx, const T *box,
 
       r2 += rij * rij;
     }
-    *output++ = sqrt(r2);
+    *output++ = std::sqrt(r2);
+  }
+}
+
+template <typename T>
+void VanillaCalcAngles(const T *coords1, const T *coords2, const T *coords3,
+                       const T *box, unsigned int nvals, T *output) {
+  T rji[3];
+  T rjk[3];
+  T xp[3];
+  for (unsigned int i = 0; i < nvals; ++i) {
+    for (unsigned char j = 0; j < 3; ++j) {
+      T temp = coords1[i * 3 + j] - coords2[i * 3 + j];
+      T adj = std::round(temp / box[j]);
+      rji[j] -= adj * box[j];
```

Comment:
Doesn't this need to be initialised to zero?

Reply:
Which one sorry?

Reply:
I am an idiot